apps/browser-proxy/src/pg-dump-middleware/pg-dump-middleware.ts

```diff
@@ -0,0 +1,83 @@
+import ExpiryMap from 'expiry-map'
+import type { ClientParameters } from 'pg-gateway'
+import { isGetExtensionsQuery, patchGetExtensionsResult } from './get-extensions-query.ts'
+import {
+  isGetExtensionMembershipQuery,
+  patchGetExtensionMembershipResult,
+} from './get-extension-membership-query.ts'
+import { FIRST_NORMAL_OID } from './constants.ts'
+
+type ConnectionId = string
+
+const state = new ExpiryMap<ConnectionId, State>(1000 * 60 * 5)
+
+type State =
+  | { step: 'wait-for-get-extensions-query' }
+  | { step: 'get-extensions-query-received' }
+  | { step: 'wait-for-get-extension-membership-query'; vectorOid: string }
+  | { step: 'get-extension-membership-query-received'; vectorOid: string }
+  | { step: 'complete' }
+
+export function pgDumpMiddleware(
```

Comment:
Let's add a comment to this middleware explaining why we need it and what it's doing